src/main/scala/com/typesafe/sbt/packager/archetypes/jlink/JlinkPlugin.scala

```diff
@@ -0,0 +1,106 @@
+package com.typesafe.sbt.packager.archetypes
+package jlink
+
+import scala.sys.process.{Process, ProcessBuilder}
+import sbt._
+import sbt.Keys._
+import com.typesafe.sbt.SbtNativePackager.{Debian, Universal}
+import com.typesafe.sbt.packager.Keys.{bundledJvmLocation, packageName}
+import com.typesafe.sbt.packager.Compat._
+import com.typesafe.sbt.packager.archetypes.jlink._
+import com.typesafe.sbt.packager.archetypes.scripts.BashStartScriptKeys
+import com.typesafe.sbt.packager.universal.UniversalPlugin
+
+/**
+  * == Jlink Application ==
+  *
+  * This class contains the default settings for creating and deploying an
+  * application as a runtime image using the standard `jlink` utility.
+  *
+  * == Configuration ==
+  *
+  * This plugin adds new settings to configure your packaged application.
+  * The keys are defined in [[com.typesafe.sbt.packager.archetypes.jlink.JlinkKeys]]
+  *
+  * @example Enable this plugin in your `build.sbt` with
+  *
+  * {{{
+  *  enablePlugins(JlinkPlugin)
+  * }}}
+  */
+object JlinkPlugin extends AutoPlugin {
+
+  object autoImport extends JlinkKeys
+
+  import autoImport._
+
+  override def requires = JavaAppPackaging
+
+  override lazy val projectSettings: Seq[Setting[_]] = Seq(
+    target in jlinkBuildImage := target.value / "jlink" / "output",
+    jlinkBundledJvmLocation := "jre",
+    bundledJvmLocation := Some(jlinkBundledJvmLocation.value),
+    jlinkOptions := (jlinkOptions ?? Nil).value,
+    jlinkOptions ++= {
+      val log = streams.value.log
+      val run = runJavaTool(javaHome.in(jlinkBuildImage).value, log) _
+
+      val paths = fullClasspath.in(Compile).value.map(_.data.toString)
```

Comment:
Can we rely on the `toString()` method here? Does it produce proper paths for unix/windows systemsß

Reply:
`File.toString` seems to produce a valid path as long as the object itself represents a valid path - see its [documentation](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#toString()). There should be no escaping issues since `Process` should not require escaping. I'm not sure what other problems we can encounter here.

Reply:
I would rather use `getPath()` to make this explicit. `toString` may change (as well as the docs)

Reply:
Fair point. Done.